### PR TITLE
py-psycopg2: add v2.9.6

### PR DESCRIPTION
**Pull request by adamjstewart**
Supports newer Python

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-psycopg2/package.py [hunk 1]
```diff
@@ -12,14 +12,26 @@ class PyPsycopg2(PythonPackage):
...
-    depends_on("python@3.6:3.9", type=("build", "link", "run"), when="@2.9:")
-    depends_on("python@2.7:2.8,3.4:3.8", type=("build", "link", "run"), when="@2.8")
-    depends_on("python@2.6:2.8,3.2:3.7", type=("build", "link", "run"), when="@:2.7")
+    # https://github.com/psycopg/psycopg2/blob/master/doc/src/install.rst
+    # https://www.psycopg.org/docs/news.html#news
+    # https://pypi.org/project/psycopg2/#history
+    depends_on("python@:3.11", when="@2.9.5:", type=("build", "link", "run"))
+    depends_on("python@:3.10", when="@2.9.1:2.9.4", type=("build", "link", "run"))
+    depends_on("python@:3.9", when="@2.8.6:2.9.0", type=("build", "link", "run"))
+    depends_on("python@:3.8", when="@2.8.4:2.8.5", type=("build", "link", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-psycopg2/package.py:27]:
```suggestion
    depends_on("python@:3.10", when="@2.9.2:2.9.4", type=("build", "link", "run"))
    depends_on("python@:3.9", when="@2.9:2.9.1", type=("build", "link", "run"))
    depends_on("python@:3.8", when="@2.8.4:2.8.6", type=("build", "link", "run"))
```
According to the install.rst:
- 2.9.1 has a python upper bound of 3.9 (see https://github.com/psycopg/psycopg2/blob/2_9_1/doc/src/install.rst)
- 2.8.6 has a python upper bound of 3.8 (see https://github.com/psycopg/psycopg2/blob/2_8_6/doc/src/install.rst)

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-psycopg2/package.py:27]: 2.9.1 has [3.10 wheels](https://pypi.org/project/psycopg2/2.9.1/#files)

2.8.6 has [3.9 wheels](https://pypi.org/project/psycopg2/2.8.6/#files)

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-psycopg2/package.py:27]: I'm fine with changing this to match the docs if you want, but PyPI is where I got these specific versions from.

Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-psycopg2/package.py:27]: No, it's fine. I did not see the wheels and did not understand that, with the comment to pypi, you meant that.

**Review by manuelakuhn — APPROVED** [review]
LGTM and builds with `--test` on Fedora 37 with `gcc@12.2.1`

**Review by alalazo — APPROVED** [review]

**Pull request merged**